pymatgen/io/vasp/inputs.py

```diff
@@ -964,29 +960,22 @@ def __add__(self, other):
 
     def check_params(self):
         """
-        Raises a warning for nonsensical or non-existent INCAR tags and
-        parameters. If a keyword doesn't exist (e.g. there's a typo in a
-        keyword), your calculation will still run, however VASP will ignore the
-        parameter without letting you know, hence why we have this Incar method.
+        Raise a warning for nonexistent INCAR tags or invalid values.
+        If a tag doesn't exist (e.g. typo), calculation will still run,
+        however VASP will ignore the tag without letting you know.
         """
-        for key, val in self.items():
-            # First check if this parameter even exists
-            if key not in incar_params:
-                warnings.warn(f"Cannot find {key} in the list of INCAR flags", BadIncarWarning, stacklevel=2)
+        for tag, val in self.items():
+            # First check if this tag exists
+            if tag not in incar_params:
+                warnings.warn(f"Cannot find {tag} in the list of INCAR tags", BadIncarWarning, stacklevel=2)
 
-            if key in incar_params:
-                if type(incar_params[key]).__name__ == "str":
-                    # Now we check if this is an appropriate parameter type
-                    if incar_params[key] == "float":
-                        if not type(val) not in ["float", "int"]:
-                            warnings.warn(f"{key}: {val} is not real", BadIncarWarning, stacklevel=2)
-                    elif type(val).__name__ != incar_params[key]:
-                        warnings.warn(f"{key}: {val} is not a {incar_params[key]}", BadIncarWarning, stacklevel=2)
+            # Now check if the tag type is appropriate
+            elif isinstance(incar_params[tag], str) and type(val).__name__ != incar_params[tag]:
+                warnings.warn(f"{tag}: {val} is not a {incar_params[tag]}", BadIncarWarning, stacklevel=2)
 
-                # if we have a list of possible parameters, check
-                # if the user given parameter is in this list
-                elif type(incar_params[key]).__name__ == "list" and val not in incar_params[key]:
-                    warnings.warn(f"{key}: Cannot find {val} in the list of parameters", BadIncarWarning, stacklevel=2)
+            # Check if the given value is in the list
+            elif isinstance(incar_params[tag], list) and val not in incar_params[tag]:
+                warnings.warn(f"{tag}: Cannot find {val} in the list of values", BadIncarWarning, stacklevel=2)
```

Comment:
I tried to clean up the INCAR check method here. Please help me check @janosh  as I'm not quite confident if changing the unit test would break anything. Several things to clarify here:

1. I tried to use "tag" to refer to the name of keys in INCAR files, trying to be consistent with [VASP wiki](https://www.vasp.at/wiki/index.php/Category:INCAR_tag). Prior to this, a mixture of "key", "parameter", "flag" and "tag" are used and was quite confusing.

2. The checking mechanism for real numbers is buggy and unnecessary (and the checking is missing in unit test), (plus number ) so I merged it to a type check.